src/state/combat.js

```diff
@@ -209,7 +211,7 @@ const combat = {
 			dispatch("playerMob/addHealth", ITEMS[food.itemId].healAmount, { root: true });
 
 			var food = rootState["inventory"].equipment.food;
-			food.count -= 1;
+			if(Math.random()*100 > rootGetters["upgrades/get"]("foodSavingRoll")*5) food.count -= 1;

```

Comment:
It might be nice to show a quick toast here
```suggestion
			if(Math.random()*100 > rootGetters["upgrades/get"]("foodSavingRoll")*5) {
				food.count -= 1;
			} else {
				EventBus.$emit("toast", { icon: ITEMS[food.itemId].icon, text: `Food preserved!` });
			}
```